### PR TITLE
Fix/limit examples to 3 azs

### PR DESCRIPTION
**Pull request by naris-silpakit**
### What does this PR do?

Limits examples to deploy to only 3 availability zones in a given region. Previously examples would create subnets in all availability zones in a region, which did not match the description in their README.md files.

Passed e2e tests.

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Comment by jplock** [comment]
Why don't we move `slice(data.aws_availability_zones.available.names, 0, 3)` into a local variable and then reference that instead of it being called multiple times in each example?

File: examples/advanced/live/preprod/eu-west-1/application_acct/dev/main.tf [hunk 1]
```diff
@@ -74,10 +74,10 @@ module "aws_vpc" {
 
   name = local.vpc_name
   cidr = local.vpc_cidr
-  azs  = data.aws_availability_zones.available.names
+  azs  = slice(data.aws_availability_zones.available.names, 0, 3)
```
Comment (kcoleman731) [inline examples/advanced/live/preprod/eu-west-1/application_acct/dev/main.tf:77]: can you use these vars in the `public_subnets` and `private_subnets` var loops?

**Comment by naris-silpakit** [comment]
@jplock @kcoleman731 Good point! It's already done in one example so why not do that for the rest of them as well. I'll push those changes shortly.

**Review by vara-bonthu — APPROVED** [review]
LGTM 👍🏼

**Review by askulkarni2 — APPROVED** [review]
Thanks for the PR. LGTM!

**Pull request merged**